gio/src/subclass/file_monitor.rs

```diff
@@ -0,0 +1,56 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::{prelude::*, subclass::prelude::*, translate::*};
+
+use crate::{ffi, FileMonitor};
+
+// Support custom implementation of virtual functions defined in `gio::ffi::GFileMonitorClass`.
+pub trait FileMonitorImpl: ObjectImpl + ObjectSubclass<Type: IsA<FileMonitor>> {
+    fn cancel(&self) -> bool {
+        let res = self.parent_cancel();
+        // cancel should always return true as specified in documentation
+        // https://docs.gtk.org/gio/vfunc.FileMonitor.cancel.html
+        debug_assert!(res, "FileMonitor.cancel should always return true");
+        res
+    }
+}
+
+// Support parent implementation of virtual functions defined in `gio::ffi::GFileMonitorClass`.
+pub trait FileMonitorImplExt: FileMonitorImpl {
```

Comment:
You probably want to provide something here for emitting the changed signal in a useful way. And probably also provide changed as a vfunc?